grails-app/controllers/org/pih/warehouse/inventory/InventoryController.groovy

```diff
@@ -531,9 +531,18 @@ class InventoryController {
 
     def listInStock = {
         def location = Location.get(session.warehouse.id)
-        def inventoryItems = dashboardService.getInStock(location)
+        List<Category> categories = Category.findAllByIdInList(params.list("categories"))
 
-        if (params.format == "csv") {
+        // When accessing the page for the first time, the flag should be set to true
+        // Initially there not parameter button, only after running the report manually it is set
+        params.includeSubcategories = params.button ? params.includeSubcategories : true
+        if (params.includeSubcategories) {
+            categories = inventoryService.getExplodedCategories(categories)
+        }
```

Comment:
Maybe it is a good idea to create an additional function to handle this code, this part of code is duplicated multiple times. Additionally, if it is possible in Groovy to pass a function as an argument you can add calculating inventory items to the function (I am not sure if it is possible)

Reply:
Yes, you guys are right about code duplication and additionally, I noticed that I have missed more such lists that use the common `list.gsp`.
From what I can see it is possible to call the method based on the string value like [(docs)](http://www.groovy-lang.org/metaprogramming.html#_dynamic_method_names)
```groovy
dashboardService."somemethof"(args)

````
I think it would be best to do that but I am not sure where such a helper method should be. I don;t think we have ever done such a case for a controller. Any ideas?
@alannadolny @kchelstowski

Reply:
I would generally move most of this logic to the service, especially the fetching calls, but since we are tight on time, I think it could stay.
I'd recommend moving it to an InventoryService, but I don't have a quick idea of how to name the method